airbyte/_processors/sql/bigquery.py

```diff
@@ -0,0 +1,204 @@
+        self._credentials: service_account.Credentials | None = None
+        self._schema_exists: bool | None = None
+        super().__init__(cache, file_writer)
+
+    @final
+    @overrides
+    def _fully_qualified(
+        self,
+        table_name: str,
+    ) -> str:
+        """Return the fully qualified name of the given table."""
+        return f"`{self.cache.schema_name}`.`{table_name!s}`"
+
+    @final
+    @overrides
+    def _quote_identifier(self, identifier: str) -> str:
+        """Return the identifier name as is. BigQuery does not require quoting identifiers"""
+        return f"{identifier}"
+
+    @final
+    @overrides
+    def _get_telemetry_info(self) -> CacheTelemetryInfo:
+        return CacheTelemetryInfo("bigquery")
+
+    def _write_files_to_new_table(
+        self,
+        files: list[Path],
+        stream_name: str,
+        batch_id: str,
+    ) -> str:
+        """Write a file(s) to a new table.
+
+        This is a generic implementation, which can be overridden by subclasses
+        to improve performance.
+        """
+        temp_table_name = self._create_table_for_loading(stream_name, batch_id)
+
+        # Specify the table ID (in the format `project_id.dataset_id.table_id`)
+        table_id = f"{self.cache.project_name}.{self.cache.dataset_name}.{temp_table_name}"
+
+        # Initialize a BigQuery client
+        client = bigquery.Client(credentials=self._get_credentials())
+
+        for file_path in files:
+            with Path.open(file_path, "rb") as source_file:
+                load_job = client.load_table_from_file(  # Make an API request
+                    file_obj=source_file,
+                    destination=table_id,
+                    job_config=bigquery.LoadJobConfig(
+                        source_format=bigquery.SourceFormat.NEWLINE_DELIMITED_JSON,
+                        schema=[
+                            bigquery.SchemaField(name, field_type=str(type_))
+                            for name, type_ in self._get_sql_column_definitions(
+                                stream_name=stream_name
+                            ).items()
+                        ],
+                    ),
+                )
+            _ = load_job.result()  # Wait for the job to complete
+
+        return temp_table_name
+
+    def _ensure_schema_exists(
+        self,
+    ) -> None:
+        """Ensure the target schema exists.
+
+        We override the default implementation because BigQuery is very slow at scanning schemas.
+
+        This implementation simply calls "CREATE SCHEMA IF NOT EXISTS" and ignores any errors.
+        """
+        if self._schema_exists:
+            return
+
+        sql = f"CREATE SCHEMA IF NOT EXISTS {self.cache.schema_name}"
+        try:
+            self._execute_sql(sql)
+        except Exception as ex:
+            # Ignore schema exists errors.
+            if "already exists" not in str(ex):
+                raise
+
+        self._schema_exists = True
```

Comment:
This implementation doesn't check - it just runs 'CREATE IF NOT EXISTS ...'. And storing the marker "True" in `self._schema_exists` means that at most we'll run this once per cache object.